Liber Usualis - mei3/liberupdatev5.py

```diff
@@ -0,0 +1,240 @@
+import lxml.etree as ET
+
+def main(filename: str) -> None:
+    """
+    Converts an MEI file to MEI 5.0 format.
+
+    Args:
+        filename: The path to the MEI file to be converted.
+    """
+    ET.register_namespace("mei", "http://www.music-encoding.org/ns/mei")
+
+    tree = ET.parse(filename)
+    root = tree.getroot()
+
+    # Dictionaries for mapping attribute values
+    div_dict = {"minor": "maior", "minior": "maior", "major": "maxima", "final": "finalis", "small": "minima", "comma": "virgula"}
+    epi_dict = {"horizontal": "h", "vertical": "v"}
+
+    # Flags for tracking certain conditions
+    theres_an_episema = False
+    theres_a_pb = False
+    theres_a_sb = False
+    nc_flag = False
+    neume_flag = False
+
+    # Extra attributes for neumes and notes
+    extra_nc = {}
+    extra_neume = {}
+
+    # Set MEI version to 5.0
+    root.attrib["meiversion"] = "5.0"
+
+    # Iterate over all elements in the tree
+    for child in root.iter("*"):
+        child.tag = ET.QName(child).localname
+
+        # Handle graphic elements (e.g., change href to target)
+        if child.tag.endswith("graphic"): #changes some stuff in the headers
+            for att in list(child.attrib):
+                if att.endswith("href"):                    
+                    child.attrib('target', child.attrib.pop(att))
+        
+        # Handle page breaks
+        elif child.tag.endswith("pb"): #adds in pagebreak info from layout element (which gets deleted later)
+            child.tag = "peeb"
+            for att in child.attrib:                
+                if att.endswith("pageref"):                    
+                    pageref = child.attrib.pop(att)
+                    break
+            for child_2 in root.iter():
+              for attr, value in child_2.attrib.items():               
+                if attr.endswith("id") and value == pageref:
+                        child.set("n", child_2.attrib["n"])
+                        break                 
```

Comment:
This could be a `find` call, rather than iterating through the whole tree again:

```python
root.find(".//meins:page[@xmlns:id='pageref']", namespaces = ...)
```